src/main/java/io/pravega/connector/boomi/PravegaWriteOperation.java

```diff
@@ -139,6 +136,9 @@ private static String inputStreamToUtf8String(InputStream is) throws IOException
                 baos.write(buffer, 0, c);
             }
             return new String(baos.toByteArray(), StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            logger.log(Level.WARNING, "unable to convert input stream to string ", e);
+            return null;
```

Comment:
we should bubble up some kind of exception here - returning null may have unpredictable consequences.  Let's wrap `e` in a RuntimeException and throw that.